MentorMatchmaker/MatchMaking.fs

```diff
@@ -0,0 +1,134 @@
+﻿module MentorMatchmaker.MatchMaking
+
+open MentorMatchmaker
+open Flips
+open Flips.Types
+open Flips.SliceMap
+
+let private toTopicWeight topic : float =
+    let Anything = 1
+    let Common = 3
+    let Popular = 5
+    let Rare = 10
+
+    match topic with
+    | IntroductionToFSharp -> Common
+    | DeepDiveInFSharp -> Popular
+    | ContributeToOpenSource -> Popular
+    | WebDevelopment -> Popular
+    | ContributeToCompiler -> Rare
+    | MachineLearning -> Rare
+    | DistributedSystems -> Rare
+    | MobileDevelopment -> Rare
+    | DomainModelling -> Popular
+    | UpForAnything -> Anything
+    | DesigningWithTypes -> Rare
+    | MetaProgramming -> Rare
+
+type private ApplicantData = {
+    Id: int
+    Applicant: Applicant
+    Topics: Set<Topic>
+    WeekTimes: Set<WeekTime>
+}
+
+type private MentorId = int
+type private MenteeId = int
+
+let private toApplicantData id applicant =
+    { Id = id
+      Applicant = applicant
+      WeekTimes = toWeekTimes applicant.Availabilities
+      Topics = applicant.Topics }
+
+type UnpairedApplicants = ApplicantPool
+
+let matchApplicants (applicantPool:ApplicantPool) : MentorshipPair list * UnpairedApplicants =
+    let mentors, mentees =
+        let mutable i = 0
+        ( [ for mentor in applicantPool.Mentors do toApplicantData i mentor; i <- i + 1 ],
+          [ for mentee in applicantPool.Mentees do toApplicantData i mentee; i <- i + 1 ] )
+
+    let matchablePairs = [
+        for mentor in mentors do
+            for mentee in mentees do
+                if (Set.overlaps mentor.Topics mentee.Topics || mentor.Topics.Contains UpForAnything || mentee.Topics.Contains UpForAnything)
+                   && Set.overlaps mentor.WeekTimes mentee.WeekTimes then
```

Comment:
I'd refactor it into a bool function that will make it easier to debug in the future :)

Reply:
What do you think of the additional line breaks? If it's still no good I'll move it to a variable.

Reply:
https://github.com/Kavignon/FSharp-Mentorship-Automatic-Planner/pull/26/commits/150c10b1981626f388ff87873fe982cbde199505#diff-c7a7aeda72bcdf556f2db7aad737bb722ac9f52537d74ab5db80183f96d2c698R70